### ISSUE TITLE
Pre-built image isn't working

### ISSUE THREAD

**Issue by bewiwi** [ovh/terraform-ovh-publiccloud-k8s#2]
We can't set *post_install_modules* to false.

install_ids is referenced in output. This problem seems to be present in depending module too

````
Error: Error applying plan:

3 error(s) occurred:

* module.k8s.module.post_install_k8s.output.install_ids: Resource 'null_resource.post_install_k8s' does not have attribute 'id' for variable 'null_resource.post_install_k8s.*.id'
* module.k8s.module.post_install_cfssl.output.install_id: Resource 'null_resource.post_install' does not have attribute 'id' for variable 'null_resource.post_install.*.id'
* module.k8s.module.post_install_etcd.output.install_ids: Resource 'null_resource.post_install_etcd' does not have attribute 'id' for variable 'null_resource.post_install_etcd.*.id'

````

**Comment by yanndegat** [comment]
yo @bewiwi 

many things here:

- i'll add the packer glance example in the next days.
- these "tf code blocks" work in other modules, so i'm pretty confident i can make them work when i'll add the packer part.
- YET, these are over complex code blocks, definately not respecting the terraform philosophy, 
which is kind of my specialty: making dirty over complex code blocks.

so i think we should have a little debate here:

shall we remove these blocks instead of fixing them?

but before answering this question with the most obvious answer... let me add some context
on why these code blocks popped up:

Terraform builds a dependency graph of the resources it has to build, then walks this path & try
to parallelize steps.

Thus, when i build i network, add a nat gateway, and add instances on a private network that may 
try to download things from the internet (mainly from install-scripts, but it could also be the case
in userdata scripts), it appears that the nat gateway may be active after the private instances.

and it results in failed install scripts/userdata scripts.

so i had to find a way to fix this issue. & i believe i've taken a wrong path.
i thought: i have to be sure that the nat is up before the scripts are run. thus, i have to introduce
a bootstrap order, and tweak the terraform depedency graph with the code blocks you've been referring to.
As when i've come up with a dirty code block i love to copy/paste it everywhere, i've added this tweak
every where it could be useful :)) namely in cluster workloads such as consul/etcd/...  etc.

So, as a first step, maybe we could remove these code blocks everywhere but the network nat gateway.

then fix the whole thing by reworking install-scripts so that they introduce "access to internet is up" as a 
dependency. so that bootstrap & post provisionning don't fail & wait until internet is up with a parameterized timeout.


what do you think?

**Comment by bewiwi** [comment]
Let me resume to be sure I understand well. 
Theses output block was only use when you want to use a GW, correct ?

If it's the case, why just booting other instances after GW is ok ? and remove all blocks in other modules. Is it what you propose saying that ?
````
then fix the whole thing by reworking install-scripts so that they introduce "access to internet is up" as a dependency
````

**Comment by yanndegat** [comment]
that's it. 
i think i have to add a private network example to be sure exactly what block can be safely removed
& i'll add the glance image build afterwards & fix network / etcd / cfssl modules accordingly

**Comment by yanndegat** [comment]
fixed by #56
